src/ResourceManager/Network/Commands.Network/NetworkWatcher/ConnectionMonitor/NewAzureNetworkWatcherConnectionMonitorCommand.cs

```diff
@@ -0,0 +1,224 @@
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the network watcher resource group.",
+            ParameterSetName = "SetByName")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Location of the network watcher.",
+            ParameterSetName = "SetByLocation")]
+        [LocationCompleter("Microsoft.Network/networkWatchers/connectionMonitors")]
+        [ValidateNotNull]
+        public string Location { get; set; }
+
+        [Alias("ConnectionMonitorName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The connection monitor name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The ID of the connection monitor source.")]
+        [ValidateNotNullOrEmpty]
+        public string SourceResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Monitoring interval in seconds.")]
+        [ValidateNotNullOrEmpty]
+        public int? MonitoringIntervalInSeconds { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Source port.")]
+        [ValidateNotNull]
+        [ValidateRange(1, int.MaxValue)]
+        public int SourcePort { get; set; }
+
+        [Parameter(
+             Mandatory = false,
+             HelpMessage = "The ID of the connection monitor destination.")]
+        [ValidateNotNullOrEmpty]
+        public string DestinationResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Ip address of the connection monitor destination.")]
+        [ValidateNotNullOrEmpty]
+        public string DestinationAddress { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Destination port.")]
+        [ValidateNotNull]
+        [ValidateRange(1, int.MaxValue)]
+        public int DestinationPort { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Configure connection monitor, but do not start it")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter ConfigureOnly { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable which represents resource tags.")]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Do not ask for confirmation if you want to overwrite a resource")]
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public string resourceGroupName = string.Empty;
```

Comment:
@irrogozh I would move these variables to under the `base.Execute()` call and instantiate their values to be `this.ResourceGroupName` and `this.NetworkWatcherName`, respectively. Then you can remove the if-else block for the `SetByName` parameter set

Reply:
Done